transport/pool.go

```diff
@@ -0,0 +1,252 @@
+package transport
+
+import (
+	"errors"
+	"math"
+	"math/rand"
+	"runtime"
+	"time"
+
+	"scylla-go-driver/frame"
+	. "scylla-go-driver/frame/request"
+	. "scylla-go-driver/frame/response"
+
+	"go.uber.org/atomic"
+)
+
+// Node Connection Pool - One per node, Node struct holds pointer to it
+// This struct holds connection to shards along with helper variables.
+type ConnPool struct {
+	conns     []atomic.Value
+	alertChan chan uint16
+
+	nrShards  uint16
+	msbIgnore uint8
+
+	idx atomic.Uint64
+}
+
+// Atomicly stores pointer to a connection.
+func (c *ConnPool) storeConn(idx uint16, conn *Conn) {
+	c.conns[idx].Store(conn)
+}
+
+// Atomicly load pointer to a connection.
+func (c *ConnPool) loadConn(idx uint16) *Conn {
+	conn := c.conns[idx].Load()
+	if conn == nil {
+		return nil
+	}
+	return conn.(*Conn)
+}
+
+// RandConn returns a random connection but has no protection
+// of it being nil, that's why NextConn() might be a better choice for now.
+func (c *ConnPool) RandConn() *Conn {
+	idx := rand.Uint32() % uint32(c.nrShards)
+	return c.loadConn(uint16(idx))
+}
+
+// NextConn returns nearest (from idx % nrShards) not nil connection
+// if it iterates over whole slice and doesn't find any (extremely unlikely) it returns nil.
```

Comment:
No need to describe implementation, it's effectively plain round robin, can return nil if pool is empty.